core/shared/src/main/scala/zio/prelude/fx/Cause.scala

```diff
@@ -0,0 +1,187 @@
+package zio.prelude.fx
+
+import zio.prelude._
+
+import scala.annotation.tailrec
+
+/**
+ * `Cause` is a data type that represents the potentially multiple ways that a
+ * computation can fail.
+ */
+sealed trait Cause[+E] { self =>
```

Comment:
This is more general than just failure. Especially if we add an empty, it becomes a free semiring. It maybe should live in `zio.prelude` directly and not speak about errors, just "events" or something, in parallel and sequence. It can be used for errors but could be used for tests and other things. Anywhere you need a tree of parallel / sequential operations.

Reply:
Yes, we could definitely generalize it and then just have `Cause` be a type alias for it.